dpypelines/pipeline/dataset_ingress_v1.py

```diff
@@ -1,22 +1,92 @@
+
+    Raises:
+        ValueError: If required files, supplementary distributions, or pipeline configuration are not found in the input directory.
+        Exception: If any other unexpected error occurs.
     """
 
-    # verify that the specified required files have been provided
+    # Attempt to access the local data store
+    try:
+        local_store = LocalDirectoryStore(files_dir)
+    except Exception as err:
+        notify.data_engineering(message.unexpected_error(f"Failed to access local data at {files_dir}", err))
+        raise
+
+    # Load the pipeline configuration as a dictionary
+    try:
+        pipeline_config: dict = local_store.get_lone_matching_json_as_dict("pipeline-config.json")
+    except Exception as err:
+        notify.data_engineering(message.pipeline_input_exception({"pipeline-config.json": "expected"}, local_store, err))
+        raise
+
+    
+    # Make sure pipeline_config contains a "pipeline" key
+    try:
+        pipeline = get_pipeline_identifier_from_config(pipeline_config)
+    except KeyError:
+        notify.data_engineering(message.expected_config_key_missing("Pipeline key not found in config", "pipeline", "dataset_ingress_v1"))
+        raise ValueError(message.expected_config_key_missing("Pipeline key not found in config", "pipeline", "dataset_ingress_v1"))
+    
+    # Check for the existence of a pipeline configuration file
+    try:
+        if not local_store.has_lone_file_matching("pipeline-config.json"):
+            notify.data_engineering(message.expected_local_file_missing("Pipeline config not found", "pipeline-config.json", "dataset_ingress_v1"))
+            raise ValueError(message.expected_local_file_missing("Pipeline config not found", "pipeline-config.json", "dataset_ingress_v1"))
+    except Exception as err:
+        notify.data_engineering(message.unexpected_error("Failed to check for pipeline config", err))
+        raise
+
+    # Extract the patterns for required files from the pipeline configuration
+    try:
+        required_file_patterns = matching.get_required_files_patterns(pipeline_config)
+    except Exception as err:
+        notify.data_engineering(message.unexpected_error("Failed to get required files patterns", err))
+        raise
+
+    # Check for the existence of each required file
+    for required_file in required_file_patterns:
+        try:
+            if not local_store.has_lone_file_matching(required_file):
+                notify.data_engineering(message.expected_local_file_missing(f"Required file {required_file} not found", required_file, "dataset_ingress_v1"))
+                raise ValueError(message.expected_local_file_missing(f"Required file {required_file} not found", required_file, "dataset_ingress_v1"))
+        except Exception as err:
+            notify.data_engineering(message.unexpected_error(f"Failed to check for required file {required_file}", err))
```

Comment:
"Error while looking for required file...."

We didn't fail to check, we checked and it went boom 😄